### PR TITLE
Anubisath Defender EventAI adjustements

### PR DESCRIPTION
**Pull request by balakethelock**
## 🍰 Pullrequest

- Removes the aggro and social pull Events.
- Adjusted spell timers based on sniff data.
- Simplified the script to only use 4 phases
- Implemented dropping Obsidian Shard on death
- Adjusted the range of Shadow Storm spells to 25 yards. The tooltip says 25 yards, and I haven't seen any evidence from classic discords, raid guides, wiki, videos to suggest that it should be less than that.

### Proof

- Proof that aggro & social pull events are not correct to classic:
https://www.youtube.com/watch?v=eWPnt_Lersc&t=1018s
https://www.youtube.com/watch?v=RF1hE_YFx4g&t=901s

### Issues

- None

### How2Test

- none

### Todo / Checklist

- [X] None

### CODE REVIEW

File: sql/migrations/20240120155120_world.sql [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+(1527701, 0, 0, 46, 1, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Anubisath Defender - Random Phase between 1 and 4'),
+(1527701, 0, 0, 49, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Anubisath Defender - Combat Pulse');
+
+DELETE FROM `creature_ai_scripts` WHERE `id`=1527702;
+INSERT INTO `creature_ai_scripts` (`id`, `delay`, `priority`, `command`, `datalong`, `datalong2`, `datalong3`, `datalong4`, `target_param1`, `target_param2`, `target_type`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `condition_id`, `comments`) VALUES
+(1527702, 0, 0, 15, 27630, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Anubisath Defender - Drop Obsidian');
+
+DELETE FROM `creature_ai_scripts` WHERE `id`=1527703;
+INSERT INTO `creature_ai_scripts` (`id`, `delay`, `priority`, `command`, `datalong`, `datalong2`, `datalong3`, `datalong4`, `target_param1`, `target_param2`, `target_type`, `data_flags`, `dataint`, `dataint2`, `dataint3`, `dataint4`, `x`, `y`, `z`, `o`, `condition_id`, `comments`) VALUES
+(1527703, 0, 0, 15, 26555, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Anubisath Defender - Cast Thunderclap (Phase 1+2)');
```
Comment (ratkosrb) [inline sql/migrations/20240120155120_world.sql]: I think you made a mistake here. This is the spell id of Shadow Storm yet comment says Thunderclap.

**Comment by ratkosrb** [comment]
By the way, looking at the list of spell ids that the creature has been seen casting, not all seem to be used in your script. I've highlighted the ones missing from the script here.

Untitled

**Comment by ratkosrb** [comment]
Spell 25699 seems to be triggered by 25698 which is cast in the script, so ignore that one.

**Comment by balakethelock** [comment]
> By the way, looking at the list of spell ids that the creature has been seen casting, not all seem to be used in your script. I've highlighted the ones missing from the script here.

These spells are dummy effects, I was faced with the choice of either implementing them as blizzard probably did which would require hardcoding their effects in spelleffects.cpp, or just go with EventAI

I decided to apply [Vmangos's project guidelines](https://github.com/vmangos/core#:~:text=off%20by%20default.-,DB%20Scripting,-%3A%20Content%20should%20be) and just went with the simpler method

**Comment by ratkosrb** [comment]
If the dummy spells are just for selecting spell to cast, they can still be handled in EventAI, by using EVENT_T_HIT_BY_SPELL (when we are hit by a spell) and EVENT_T_HIT_BY_AURA (when aura is applied to us) events. Or we can just have the script cast them at the appropriate time even if they are not handled, just to make it so the order of spell cast packets in vmangos is the same as on blizzard servers on the packet level. I think it would be nice to utilize them. If you don't want to do it, I can merge it as is, but it would be cooler if you did.

**Pull request merged**